stan/math/prim/fun/beta.hpp

```diff
@@ -48,7 +48,7 @@ namespace math {
  * @param b Second value
  * @return Beta function applied to the two values.
  */
-template <typename T1, typename T2>
+template <typename T1, typename T2, require_all_arithmetic_t<T1, T2>* = nullptr>
```

Comment:
[question]
I'm confused as to why we have seemingly three different ways to do this:

1.  `typename = require_all_arithmetic_t<...>`
2. `require_all_arithmetic_t<...>...`
3. `require_all_arithmetic_t<T1, T2>* = nullptr`

I'd rather just settle on one unless there's an important difference.  @SteveBronder?

Reply:
Let's stick with `require_all_arithmetic_t<T1, T2>* = nullptr`

Reply:
I'm OK with any of (1) -- (3).  Should we always use the `ptr = nullptr` idiom?  I was wondering if there were differences in the approaches.

Reply:
1. `typename = ` cannot be used for function overloads that opnly differ in requires. 
2. `...` can not be used on classes that need more than one require. Although I don't think we use requires for classes much.
3. `* = nullptr` has AFAIK no such restrictions. 

So either 2. or 3. is fine. 2. is the shortest, but so far we have used 3. the most.